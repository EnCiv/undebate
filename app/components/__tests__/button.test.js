import React from 'react'
import { shallow } from 'enzyme'

import Button from '../button'

describe('Button', () => {
  it('should render correctly', () => {
    const component = shallow(<Button />)

    expect(component).toMatchSnapshot()
  })

  it('should call onClick when button is clicked', () => {
    const onClick = jest.fn()
    const component = shallow(<Button onClick={onClick} />)

    component.simulate('click')

    expect(onClick).toHaveBeenCalled()
  })

  it('should pass className to rendered component', () => {
    const component = shallow(<Button className="foo" />)

    expect(component).toHaveClassName('foo')
  })

  it('should render child element', () => {
    const component = shallow(<Button>Next</Button>)

    expect(component.childAt(0).text()).toEqual('Next')
    expect(component).toMatchSnapshot()
  })
})
