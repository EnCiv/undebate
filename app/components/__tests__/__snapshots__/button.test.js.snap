// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render child element 1`] = `
<Button
  classes={
    Object {
      "Button": "Button-Button-0-1-1",
    }
  }
>
  Next
</Button>
`;

exports[`Button should render correctly 1`] = `
<Button
  classes={
    Object {
      "Button": "Button-Button-0-1-1",
    }
  }
/>
`;
